eth/core/precompile/stateful.go

```diff
@@ -89,16 +89,26 @@ func (sc *stateful) Run(
 		return nil, err
 	}
 
+	// Save the length of the StateDB's logs to ensure that no logs are emitted in read-only mode.
+	var logsLen int
+	if readonly {
+		logsLen = len(evm.GetStateDB().Logs())
+	}
+
 	// Execute the method registered with the given signature with the given args.
 	vals, err := method.Execute(
 		ctx,
 		evm,
 		caller,
 		value,
-		readonly,
 		unpackedArgs...,
 	)
 
+	// Ensure that no logs were emitted during the execution of the precompile if in read-only mode.
+	if readonly && len(evm.GetStateDB().Logs()) > logsLen {
```

Comment:
I think there is a nicer way to handle this using a TryLock or Sempahore or something, opposed to using log length.

Reply:
Sure